pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -58,11 +59,16 @@ func NewDockerRegistryServiceController(cl client.Interface, options DockerRegis
 			UpdateFunc: e.serviceUpdated,
 			DeleteFunc: e.serviceDeleted,
 		},
+		cache.Indexers{"namespace": cache.MetaNamespaceIndexFunc},
```

Comment:
actually, you're already scoped to a namespace, you just need the service name in the key